commands/create/nodepool/command_test.go

```diff
@@ -0,0 +1,318 @@
+package nodepool
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/afero"
+
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/testutils"
+)
+
+// TODO:
+// cases which should succeed:
+//   - setting only scaling min or only max should result in proper setting
+//   - set AZ list to letters like "A,B,C"
+
+// TestCollectArgs tests whether collectArguments produces the expected results.
+func TestCollectArgs(t *testing.T) {
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would question why we need a mock server again. I think we should try to not write tests involving protocol layers if possible.

Reply:
It's there to satisfy a call to the API's info endpoint. Any idea how to prevent or mock this otherwise?

Reply:
What do you want to test? We do not need to test TCP connections, TCP handshakes or dial or transport functions. This is all not relevant here I think. I think what we want to test is everything up to the API client call. When you want to verify behaviour based on client responses I also suggested fake transport implementations. Depends a bit on your goal. I am not saying what you do here is wrong and I very much appreciate that you have such a strong focus on automated verification of functionality. I just want to challenge the idea a little bit and see what we can do different.

Reply:
For the record, this is not a blocking topic now for this PR.